### PR TITLE
Fix #330: Markdown changelog links

### LINKED ISSUE #330: Changelog links on download pages lead to 404 due to text to markdown renames
changelog.txt was renamed to changelog.md in https://github.com/OpenTTD/OpenTTD/pull/12994

Since then the changelog link in [_layouts/download.html](https://github.com/OpenTTD/website/blob/main/_layouts/download.html) 404s for nightly builds.

### PR DESCRIPTION
**Pull request by merni-ns**
Closes #330.

Fix the links in download pages and 15.0-beta1 blogpost to point to the new changelog locations.

For 15.0-beta1, this will still point to an empty page since the actually generated changelog is broken (see OpenTTD/OpenTTD#13195).

Is there any other place where changelogs are linked?

### CODE REVIEW

**Comment by stormcone** [comment]
The `_layouts/download.html` is used not just for OpenTTD itself, but also for other files, like OpenGFX, grfcodec, etc. Those are still using plain text format for change logs. This is currently not a problem, as the latest versions of those are released before the markdown change were done, but can be for the future releases.

**Comment by TrueBrain** [comment]
> The `_layouts/download.html` is used not just for OpenTTD itself, but also for other files, like OpenGFX, grfcodec, etc. Those are still using plain text format for change logs. This is currently not a problem, as the latest versions of those are released before the markdown change were done, but can be for the future releases.

If I remember correctly, `raw_type` contains something like `openttd-nightlies` or `openttd-releases`, which can tell that it is actually an OpenTTD release. But yeah, adding this additional check would be good, just to avoid future issues :)

**Comment by auge8472** [comment]
See also #331 for further details. I was unable to solve #330 but maybe there is something interesting to read.

File: _posts/2024-12-24-openttd-15-0-beta1.md [hunk 1]
```diff
@@ -21,5 +21,5 @@ To be a bit less vague, here are some of the highlights:
 There's a lot of bugs left for you to find, though, guaranteed. Enjoy the beta, test all the things and [report anything you find amiss](https://github.com/OpenTTD/OpenTTD/issues/new/choose), to make OpenTTD 15 the best OpenTTD 15 ever.
 
 * [Download](https://www.openttd.org/downloads/openttd-releases/testing.html)
-* [Changelog](https://cdn.openttd.org/openttd-releases/15.0-beta1/changelog.txt)
+* [Changelog](https://cdn.openttd.org/openttd-releases/15.0-beta1/changelog.md)
```
Comment (TrueBrain) [inline _posts/2024-12-24-openttd-15-0-beta1.md:24]: I split this off into https://github.com/OpenTTD/website/pull/334 , as fixing the empty `changelog.md` was easy, making this change a trivial fix :) The other parts of this PR are a bit more tricky :)

File: _layouts/download.html [hunk 2]
```diff
@@ -34,7 +34,13 @@ <h3>Download {{ page.name }}</h3>
         <div class="content">
             <p>Latest {{ page.name }} release in {{ page.category }} is {{ page.version }}, released on {{ page.date | date: "%Y-%m-%d %H:%M" }} UTC.</p>
 
+            {% assign markdowndate = '22-10-2024' | date: '%s' %}
```
Comment (TrueBrain) [inline _layouts/download.html:37]: Would `2024-10-22` also work? Just to be friendly to our American friends? :)

**Review by TrueBrain — APPROVED** [review]

**Pull request merged**